tests/test_repairs.py

```
@@ -0,0 +1,1574 @@
+)
+async def test_async_create_device_id_mismatch_issue_ignores_invalid_ids(
+    monkeypatch: pytest.MonkeyPatch,
+    entry_device_id: str,
+    observed_device_id: str,
+    expect_issue: bool,
+) -> None:
+    """Do not create mismatches when configured or observed IDs are malformed."""
+    entry = _make_entry(device_id=entry_device_id)
+    called: dict[str, int] = {"count": 0}
+
+    def _capture_issue(**kwargs: Any) -> None:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)